source/FreeRTOS_IP_Utils.c

```diff
@@ -856,12 +859,22 @@ uint16_t usGenerateChecksum( uint16_t usSum,
     }
 
     /* Word (32-bit) aligned, do the most part. */
-    xLastSource.u32ptr = ( xSource.u32ptr + ( uxDataLengthBytes / 4U ) ) - 3U;
+
+    ulSize = ( uintptr_t ) ( ( uxDataLengthBytes / 4U ) * 4U );
+
+    if( ulSize >= ( 3U * sizeof( uint32_t ) ) )
+    {
+        ulSize -= ( 3U * sizeof( uint32_t ) );
+    }
+    else
+    {
+        ulSize = 0U;
+    }
 
     /* In this loop, four 32-bit additions will be done, in total 16 bytes.
      * Indexing with constants (0,1,2,3) gives faster code than using
      * post-increments. */
-    while( xSource.u32ptr < xLastSource.u32ptr )
+    for( ulX = 0U; ulX < ulSize; ulX += 16U )
```

Comment:
Please check if we can have ulX += 4U * sizeof( uint32_t ) , instead of of ulX += 16U at Line 877.